### DISCUSSION TITLE
RTOS for RPG4

### DISCUSSION THREAD

**Discussion by rockoguido** [Mi-V-Soft-RISC-V/miv-rv32-documentation#1] [RTOS for Mi-V RV32 Soft Processors] [answered]
Hi.  I have a user targeting the RPG4 looking for a RTOS running on the MiV. Told them FreeRTOS was an option and perhaps Zephyr but wanted to check.. I believe FreeRTOS is the only legit option and that Zephyr is only available for the SoC (i.e. PolarFire SoC) but wanted to verify
Thx
Dave

**Comment by nitindeshpande** [comment, marked as answer]
@rockoguido 
Thats correct Dave, with MIV_RV32 we are planning to support only FreeRTOS for now.

**Reply by TommyMurphyTM1234** [reply]
In case it's of any use, a Zephyr example for Mi-V RV32 on IGLOO2 M2GL025 exists here:

* https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/riscv/m2gl025_miv
* https://docs.zephyrproject.org/latest/boards/riscv/index.html

Maybe the user could port it to RTG4 if they really need Zephyr?

**Reply by JV73** [reply]
Can someone please point to the RV32 port for FreeRTOS ?  I did not recognize it in the latest FreeRTOS Distro examples.

**Reply by nitindeshpande** [reply]
Hi Dev, We are planning to publish the FreeRTOS on MIV_RV32 next week. I will notify here when it is pushed.

**Reply by rockoguido** [reply]
hello!  ok thanks so much, all updates are appreciated!!

**Reply by nitindeshpande** [reply]
The FreeRTOS example project is now available at [miv-rv32-freertos-demo](https://github.com/Mi-V-Soft-RISC-V/miv-rv32-bare-metal-examples/tree/main/applications/freertos/miv-rv32-freertos-demo).
